evalml/model_understanding/graphs.py

```diff
@@ -605,221 +612,250 @@ def partial_dependence(
             feature value pair.
 
     Raises:
-        ValueError: if the user provides a tuple of not exactly two features.
-        ValueError: if the provided pipeline isn't fitted.
-        ValueError: if the provided pipeline is a Baseline pipeline.
-        ValueError: if any of the features passed in are completely NaN
-        ValueError: if any of the features are low-variance. Defined as having one value occurring more than the upper
+        PartialDependenceError: if the user provides a tuple of not exactly two features.
+        PartialDependenceError: if the provided pipeline isn't fitted.
+        PartialDependenceError: if the provided pipeline is a Baseline pipeline.
+        PartialDependenceError: if any of the features passed in are completely NaN
+        PartialDependenceError: if any of the features are low-variance. Defined as having one value occurring more than the upper
             percentile passed by the user. By default 95%.
     """
 
-    # Dynamically set the grid resolution to the maximum number of values
-    # in the categorical/datetime variables if there are more categories/datetime values than resolution cells
-    X = infer_feature_types(X)
-
-    if isinstance(features, (list, tuple)):
-        is_categorical = [
-            _is_feature_of_type(f, X, ww.logical_types.Categorical) for f in features
-        ]
-        is_datetime = [
-            _is_feature_of_type(f, X, ww.logical_types.Datetime) for f in features
-        ]
-    else:
-        is_categorical = [
-            _is_feature_of_type(features, X, ww.logical_types.Categorical)
-        ]
-        is_datetime = [_is_feature_of_type(features, X, ww.logical_types.Datetime)]
-
-    if isinstance(features, (list, tuple)):
-        if len(features) != 2:
-            raise ValueError(
-                "Too many features given to graph_partial_dependence.  Only one or two-way partial "
-                "dependence is supported."
+    try:
```

Comment:
Adding this `try` unfortunately makes the diff seem bigger than it is. I think it's important to catch all possible errors that this can raise so that users don't have to resort to `except Exception` or `except PartialDependenceError, ValueError` and hope that'll catch everything.

Reply:
We have an outstanding issue to clean up partial dependence (https://github.com/alteryx/evalml/issues/2502) so maybe this will be cleaner once we redesign this with this requirement in mind.